### PR TITLE
lestarch: update Ref with new components

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| Infrastructure |
|**_Affected Component_**| Ref/Gds |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**| y  |
|**_Builds Without Errors (y/n)_**| y |
|**_Unit Tests Pass (y/n)_**| y  |
|**_Documentation Included (y/n)_**| n/a |

---
## Change Description

This updates Ref per #515 to use the latest in IPv4 Drivers.  It included the update from #538 preset for the new components.  If you still use the old GroundInterface, set checksum to use "fixed".  See below.

## Rationale

We needed to adopt the new ground components in our Ref app.  

## Testing/Review Recommendations

Tested against Gds and a file downlink of 10MB.  Works without issue.

## Future Work

1. Decide if we need both `StaticMemory` and `BufferManager` as examples or if we want to simplify by using just BufferManager.
2. Configuration for GDS (see #537) such that the checksum is not hardcoded.

Note: neither should block this PR.

### CODE REVIEW

**Comment by LeStarch** [comment]
https://github.com/nasa/fprime/blob/52fb2330039f4d44b1b7b2fb5a4e9f5128de57d3/Gds/src/fprime_gds/common/communication/checksum.py#L14 for setting the checksum.

**Review by saba-ja — COMMENTED** [review]
The Ref does compile and run without issue, but the RPI seems to have problem.
After changing the following
https://github.com/nasa/fprime/blob/4c2ce7ade1c5316c8c9a4e84508b5d568740529e/RPI/Top/Topology.cpp#L262

to `(void) comm.joinSocketTask(NULL);`

received more errors as shown below:

```
fprime/RPI/build-fprime-automatic-native/RPI/Top/RPITopologyAppAc.cpp:185:14: error: no matching member function for call to 'set_write_OutputPort'
    groundIf.set_write_OutputPort(0, comm.get_send_InputPort(0));
    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
fprime/RPI/build-fprime-automatic-native/F-Prime/Svc/GroundInterface/GroundInterfaceComponentAc.hpp:129:10: note: candidate function not viable: no known conversion from 'Drv::InputByteStreamSendPort *' to 'Fw::InputBufferSendPort *' for 2nd argument
    void set_write_OutputPort(
         ^
fprime/RPI/build-fprime-automatic-native/F-Prime/Svc/GroundInterface/GroundInterfaceComponentAc.hpp:202:10: note: candidate function not viable: no known conversion from 'Drv::InputByteStreamSendPort *' to 'Fw::InputSerializePort *' for 2nd argument
    void set_write_OutputPort(
         ^
fprime/RPI/build-fprime-automatic-native/RPI/Top/RPITopologyAppAc.cpp:317:10: error: no matching member function for call to 'set_recv_OutputPort'
    comm.set_recv_OutputPort(0, groundIf.get_readCallback_InputPort(0));
    ~~~~~^~~~~~~~~~~~~~~~~~~
fprime/RPI/build-fprime-automatic-native/F-Prime/Drv/ByteStreamDriverModel/ByteStreamDriverComponentAc.hpp:80:10: note: candidate function not viable: no known conversion from 'Fw::InputBufferSendPort *' to 'Drv::InputByteStreamRecvPort *' for 2nd argument
    void set_recv_OutputPort(
         ^
fprime/RPI/build-fprime-automatic-native/F-Prime/Drv/ByteStreamDriverModel/ByteStreamDriverComponentAc.hpp:109:10: note: candidate function not viable: no known conversion from 'Fw::InputBufferSendPort *' to 'Fw::InputSerializePort *' for 2nd argument
    void set_recv_OutputPort(
         ^
2 errors generated.
make[3]: *** [RPI/Top/CMakeFiles/RPI_exe.dir/RPITopologyAppAc.cpp.o] Error 1
make[2]: *** [RPI/Top/CMakeFiles/RPI_exe.dir/all] Error 2
make[1]: *** [RPI/Top/CMakeFiles/RPI.dir/rule] Error 2
make: *** [RPI] Error 2
[ERROR] CMake erred with return code 2
```

File: Gds/src/fprime_gds/common/communication/checksum.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    """ Initial checksum implementation for FpFramerDeframer. """
+    return zlib.crc32(data) & 0xffffffff
+
+
+CHECKSUM_SELECTION = "crc32"
+CHECKSUM_MAPPING = {
+    "fixed": lambda data: 0xcafecafe,
+    "crc32": crc_calculation,
+    "default": crc_calculation
+}
```
[the remark this answers is not in the document]
Comment (LeStarch) [reply Gds/src/fprime_gds/common/communication/checksum.py:19]: Yes, they reference the function and I was getting an crc_calculation undefined message

**Comment by LeStarch** [comment]
@saba-ja I am making this a draft.  I saw the same problems late last night, so I am going to need to work on the RPI example.

**Comment by LeStarch** [comment]
@saba-ja this is ready for review.  I have done the following:

1. Fixed Ref, RPI, and (above) review comments
2. Build and ran each with `fprime-gds`
3. Tested file up and down with each

It should be ready for review and merge!

**Review by saba-ja — APPROVED** [review]
Built and ran for both Ref and RPI with no issue.
There is still room for improvement in both RPI and Ref, but since FPP is in the near future, it might not worth the effort.
This PR should be OK to merge into devel.

**Comment by LeStarch** [comment]
I'll fix that header in a separate PR.

**Pull request merged**